nutanix/data_source_foundation_discover_nodes.go

```diff
@@ -0,0 +1,168 @@
+										Computed: true,
+									},
+									"current_network_interface": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"node_position": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"hypervisor": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"configured": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+									"nos_version": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"cluster_id": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"current_cvm_vlan_tag": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"hypervisor_version": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"svm_ip": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"model": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// dataSourceFoundationDiscoverNodesRead performs get operation on /discover_nodes api and sets it to resource data schema appropriately
+func dataSourceFoundationDiscoverNodesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// create foundation client connection
+	conn := meta.(*Client).FoundationClientAPI
+
+	resp, err := conn.Networking.DiscoverNodes(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	entities := make([]map[string]interface{}, len(*resp))
+	for k, v := range *resp {
+		entity := make(map[string]interface{})
+		entity["model"] = v.Model
+		entity["chassis_n"] = v.ChassisN
+		entity["block_id"] = v.BlockID
+
+		// construct node details map
+		entity["nodes"] = flattenDiscoveredNodes(v.Nodes)
+
+		entities[k] = entity
+	}
+
+	if setErr := d.Set("entities", entities); setErr != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(resource.UniqueId())
+	return nil
+}
+
+// flattenDiscoveredNodes constructs node details map for each discovered node
+func flattenDiscoveredNodes(nodesList []foundation.DiscoveredNode) []map[string]interface{} {
+	nodes := make([]map[string]interface{}, len(nodesList))
+	for k, v := range nodesList {
+		node := make(map[string]interface{})
+
+		node["foundation_version"] = v.FoundationVersion
+		node["ipv6_address"] = v.Ipv6Address
+		node["node_uuid"] = v.NodeUUID
+		node["current_network_interface"] = v.CurrentNetworkInterface
+		node["node_position"] = v.NodePosition
+		node["hypervisor"] = v.Hypervisor
+		node["configured"] = v.Configured
+		node["nos_version"] = v.NosVersion
+
+		// to make sure we set accept integer response
+		if v.ClusterID != nil && reflect.TypeOf(v.ClusterID).String() != "string" {
+			node["cluster_id"] = int64((v.ClusterID).(float64))
```

Comment:
can we directly convert to int64? 
Also, this can cause error too.

Reply:
According to docs for interface{} type, the underlying type after unmarshalling is float64, so it needed to be asserted with float64 then it can be typecasted to int64

Reply:
[Check this.
](https://stackoverflow.com/questions/18041334/convert-interface-to-int)

We should also handle error scenario where this conversion is not possible.